app/src/main/kotlin/pl/gov/mc/protego/ui/registration/RegistrationViewModel.kt

```diff
@@ -70,11 +71,15 @@ class RegistrationViewModel(
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
             .doOnSuccess { _sessionData.value = it }
-            .doFinally { _isInProgress.value = false }
-            .doOnSubscribe { _isInProgress.value = true }
+            .handleUiLock()
             .subscribeBy(
                 onError = { Timber.e(it, "Anonymous registration Error") },
                 onSuccess = { Timber.d("Anonymous registration request succeed") }
             ).addTo(disposables)
     }
+
+    private fun <T> Single<T>.handleUiLock() =
+        this
+            .doFinally { _isInProgress put UiLock.NO_LOCK }
```

Comment:
wydaje mi się, że ta nazwa(`inProgress`) nie do końca pasuje do przekazywanych wartości. Po nazwie spodziewałbym się informacji co jest w inProgress albo jakie są stany tego procesu. A tu są przysyłane informacje o poziomie blokowania UI. Jest jeszcze jeden aspekt, teraz view model decyduje co ma się stać na widoku. Czy nie lepiej było by gdyby to widok podejmował decyzję jak daną sytuację obsłużyć? czy to wymaga  zablokowania całego UI, a może jedynie niektórych przycisków i pół?

Reply:
Czy masz pomysł na zaadresowanie tego?